src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -780,6 +780,29 @@ public int EncodeToUtf8(Span<byte> destination)
 
         public bool Equals(Rune other) => this == other;
 
+#if SYSTEM_PRIVATE_CORELIB
```

Comment:
@stephentoub @GrabYourPitchforks are we allowed to update `Microsoft.Bcl.Memory` with the API addition here?

Reply:
We type forward Rune from that package to Rune in corelib? If so, no, we can't add any members to the type in the package. It could end up being used against an older corelib that lacks the member. @ericstj, this is still an issue, right?

Reply:
It looks like Rune was added to Microsoft.BCL.Memory in .NET 10.0 with https://github.com/dotnet/runtime/commit/c0ca2c220c8b931ee017278fdf022bb22f202f1f.  However the copy of it there is **internal**.  So you are free to chance that internal implementation, so long as it builds correctly.

The only type exposed publicly from that change to Microsoft.BCL.Memory was `Utf8`.  `Utf8` does have the constraint that @stephentoub mentioned - its surface area in Microsoft.BCL.Memory cannot be any larger than that exposed by `net8.0`, since the package does not carry the implementation for `net8.0` but it exposes a `netstandard2.0` assembly which much be compatible with `net8.0`

Reply:
@tarekgh -- I think this should answer your question but let me know.  You're fine to modify Rune in Microsoft.BCL.Memory since it's not exposed and only compiled for !netcoreapp.